Sources/SentryCrash/Recording/Monitors/SentryCrashMonitor_MachException.c

```diff
@@ -313,7 +313,30 @@ handleExceptions(void *const userData)
         kern_return_t kr = mach_msg(&exceptionMessage.header, MACH_RCV_MSG, 0,
             sizeof(exceptionMessage), g_exceptionPort, MACH_MSG_TIMEOUT_NONE, MACH_PORT_NULL);
         if (kr == KERN_SUCCESS) {
-            break;
+            // If the exception was generated in a different process than the one in which
+            // the handler was installed, ignore it.  Task-level exception handlers are
+            // inherited by subprocesses, and we don't want to handle or report exceptions
+            // in arbitrary subprocesses of the one in which we were installed.
+            if (exceptionMessage.task.name != mach_task_self()) {
+                SentryCrashLOG_DEBUG("Received exception from subprocess. Ignoring message and waiting for another.");
+                // Send a reply saying "I didn't handle this exception".
+                replyMessage.header.msgh_id = 0;
```

Comment:
`m`: I'm unsure if we can always set this to 0. I guess it's better to reference the existing id and create a new one with something like. WDYT, @armcknight?
```suggestion
                replyMessage.header.msgh_id = exceptionMessage.header.msgh_id + 1000
```

Reply:
see also https://www.gnu.org/software/hurd/gnumach-doc/Message-Format.html:
> The mach_msg call doesn't use the msgh_id field, but it conventionally conveys an operation or function id.

What happens if we keep assign the existing `exceptionMessage.header.msgh_id`?

Reply:
Seems to work fine using `exceptionMessage.header.msgh_id`.  Would you prefer that over using 0 with a comment?

Reply:
I think it makes sense to use the preexisting `msgh_id` 👍🏻

Reply:
Done.